From cc0406e5801066762dd97850754b78393bfa32b1 Mon Sep 17 00:00:00 2001
From: Rohan Puri <rohan_puri@xyratex.com>
Date: Tue, 9 Jul 2013 13:32:20 +0530
Subject: [PATCH 014/121] ADDB monitoring requires ADDB summary records to be
 sent on wire, so we define new xcode types for them in this file.

---
 addb/addb_monitor_wire.h | 63 ++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 63 insertions(+)
 create mode 100644 addb/addb_monitor_wire.h

diff --git a/addb/addb_monitor_wire.h b/addb/addb_monitor_wire.h
new file mode 100644
index 0000000..63e8b07
--- /dev/null
+++ b/addb/addb_monitor_wire.h
@@ -0,0 +1,63 @@
+/* -*- C -*- */
+/*
+ * COPYRIGHT 2012 XYRATEX TECHNOLOGY LIMITED
+ *
+ * THIS DRAWING/DOCUMENT, ITS SPECIFICATIONS, AND THE DATA CONTAINED
+ * HEREIN, ARE THE EXCLUSIVE PROPERTY OF XYRATEX TECHNOLOGY
+ * LIMITED, ISSUED IN STRICT CONFIDENCE AND SHALL NOT, WITHOUT
+ * THE PRIOR WRITTEN PERMISSION OF XYRATEX TECHNOLOGY LIMITED,
+ * BE REPRODUCED, COPIED, OR DISCLOSED TO A THIRD PARTY, OR
+ * USED FOR ANY PURPOSE WHATSOEVER, OR STORED IN A RETRIEVAL SYSTEM
+ * EXCEPT AS ALLOWED BY THE TERMS OF XYRATEX LICENSES AND AGREEMENTS.
+ *
+ * YOU SHOULD HAVE RECEIVED A COPY OF XYRATEX'S LICENSE ALONG WITH
+ * THIS RELEASE. IF NOT PLEASE CONTACT A XYRATEX REPRESENTATIVE
+ * http://www.xyratex.com/contact
+ *
+ * Original author: Rohan Puri <rohan_puri@xyratex.com>
+ * Original creation date: 07/09/2013
+ */
+
+#pragma once
+
+#ifndef __MERO_ADDB_ADDB_MONITOR_WIRE_H__
+#define __MERO_ADDB_ADDB_MONITOR_WIRE_H__
+
+#include "lib/types.h"
+#include "xcode/xcode_attr.h"
+
+/**
+    ADDB Monitor Serializable Data Types
+
+   These data types are serializable for network and stob use.
+   They are all tagged with the @ref xcode attributes that automate the
+   generation of the serialization code.
+   @{
+ */
+
+/**
+   A serializable sequence of 64 bit integers used by ADDB.
+ */
+struct m0_addb_sum_rec_seq {
+	uint32_t  asrs_nr;   /**< Sequence length */
+	uint64_t *asrs_data; /**< ADDB summary record encoded*/
+} M0_XCA_SEQUENCE;
+
+struct m0_addb_sum_rec_wire {
+	/** This is addb_rec_type:art_id for this ADDB summary record */
+	uint32_t               asrw_id;
+	/* ADDB summary record data encoded in sequence form */
+	struct m0_addb_sum_req asrw_rec;
+} M0_XCA_RECORD;
+
+#endif /* __MERO_ADDB_ADDB_MONITOR_WIRE_H__ */
+
+/*
+ *  Local variables:
+ *  c-indentation-style: "K&R"
+ *  c-basic-offset: 8
+ *  tab-width: 8
+ *  fill-column: 80
+ *  scroll-step: 1
+ *  End:
+ */
-- 
1.8.3.2

